pkg/roachprod/roachprod.go

```diff
@@ -237,7 +237,18 @@ func CachedCluster(name string) (*cloud.Cluster, bool) {
 // protects both the reading and the writing in order to prevent the hazard
 // caused by concurrent goroutines reading cloud state in a different order
 // than writing it to disk.
-func Sync(l *logger.Logger, options vm.ListOptions) (*cloud.Cloud, error) {
+//
+// overwriteMissingClusters indicates if the local cluster cache should be
+// updated to remove any clusters not returned from ListCloud. Because Azure
+// can sometimes return stale VM information, we cannot trust that missing
+// clusters are actually deleted. We should only delete clusters from the
+// cache when we know no one is using it, i.e. the start of a roachtest
+// run or a roachprod CLI invocation. This approach is best effort, it
+// is still possible for the caller to concurrently call roachprod sync
+// while a roachtest suite is running.
```

Comment:
I wonder if we could scope this behaviour _just_ for the Azure nightly run? Maybe via a command line flag that is only passed in that build.

It would help us scope this change and also make this a little more understandable. Currently, some calls get `true` and others get `false`, and it's not super clear to me why one would pass one or the other, especially in unrelated clouds.

Reply:
We chatted offline and agreed this was a good idea. I went with the `roachtest` setting a public var in `roachprod approach. It seemed slightly easier to follow than an env var.